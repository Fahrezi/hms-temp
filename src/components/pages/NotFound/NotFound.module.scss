.not_found {
  & {
    width: 100%;
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    background: linear-gradient(45deg, #1f1f38, #2c2c6c);
    overflow: hidden;
    position: relative;
  }

  &__container {
    position: relative;
    max-width: 1200px;
    width: 100%;
    padding: 0 2rem;
    z-index: 1;
  }

  &__title {
    font-size: 15rem;
    font-weight: 900;
    margin: 0;
    text-align: center;
    line-height: 1;
    color: rgba(255, 255, 255, 0.1);
    position: relative;

    @media (max-width: 768px) {
      font-size: 10rem;
    }

    @media (max-width: 480px) {
      font-size: 7rem;
    }

    &-digit {
      display: inline-block;
      animation: float 6s ease-in-out infinite;

      &:nth-child(1) {
        animation-delay: 0.2s;
      }

      &:nth-child(2) {
        animation-delay: 0.4s;
      }

      &:nth-child(3) {
        animation-delay: 0.6s;
      }
    }
  }

  &__message {
    text-align: center;
    color: #fff;
    margin-top: 2rem;

    &-title {
      font-size: 2.5rem;
      margin-bottom: 1rem;

      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    &-text {
      font-size: 1.2rem;
      max-width: 600px;
      margin: 0 auto 2rem;
      line-height: 1.6;
      color: rgba(255, 255, 255, 0.7);
    }

    &-button {
      display: inline-block;
      padding: 0.8rem 2rem;
      background: #4db5ff;
      color: #1f1f38;
      border-radius: 0.5rem;
      font-weight: 600;
      transition: all 0.3s ease;
      text-decoration: none;

      &:hover {
        background: #fff;
        box-shadow: 0 0 20px rgba(77, 181, 255, 0.4);
        transform: translateY(-3px);
      }
    }
  }

  &__stars {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    pointer-events: none;

    &::before,
    &::after {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-image:
        radial-gradient(2px 2px at 20px 30px, #fff, rgba(0, 0, 0, 0)),
        radial-gradient(2px 2px at 40px 70px, #fff, rgba(0, 0, 0, 0)),
        radial-gradient(2px 2px at 50px 160px, #fff, rgba(0, 0, 0, 0)),
        radial-gradient(2px 2px at 90px 40px, #fff, rgba(0, 0, 0, 0)),
        radial-gradient(2px 2px at 130px 80px, #fff, rgba(0, 0, 0, 0)),
        radial-gradient(2px 2px at 160px 120px, #fff, rgba(0, 0, 0, 0));
      background-repeat: repeat;
      background-size: 200px 200px;
      opacity: 0.5;
    }

    &::after {
      background-size: 300px 300px;
      background-position: 50px 50px;
      animation: stars 30s linear infinite;
    }
  }

  &__astronaut {
    position: absolute;
    top: 50%;
    right: 20%;
    width: 100px;
    height: 160px;
    animation: astronaut 15s ease-in-out infinite;
    transform: translate(-50%, -50%);

    @media (max-width: 768px) {
      display: none;
    }

    &-body {
      position: absolute;
      top: 40px;
      left: 25px;
      width: 50px;
      height: 80px;
      background: #f5f5f5;
      border-radius: 40px 40px 0 0;
    }

    &-head {
      position: absolute;
      top: 10px;
      left: 35px;
      width: 30px;
      height: 30px;
      background: #f5f5f5;
      border-radius: 50%;

      &::before {
        content: '';
        position: absolute;
        top: 5px;
        left: 5px;
        width: 20px;
        height: 15px;
        background: rgba(0, 0, 0, 0.1);
        border-radius: 10px;
      }
    }

    &-arm-left,
    &-arm-right {
      position: absolute;
      top: 50px;
      width: 15px;
      height: 40px;
      background: #f5f5f5;
      border-radius: 10px;
    }

    &-arm-left {
      left: 15px;
      transform: rotate(15deg);
      transform-origin: top center;
      animation: arm-left 4s ease-in-out infinite;
    }

    &-arm-right {
      right: 15px;
      transform: rotate(-15deg);
      transform-origin: top center;
      animation: arm-right 4s ease-in-out infinite;
    }

    &-leg-left,
    &-leg-right {
      position: absolute;
      top: 120px;
      width: 15px;
      height: 40px;
      background: #f5f5f5;
      border-radius: 0 0 10px 10px;
    }

    &-leg-left {
      left: 30px;
    }

    &-leg-right {
      right: 30px;
    }
  }
}

@keyframes float {
  0%,
  100% {
    transform: translateY(0);
  }
  50% {
    transform: translateY(-20px);
  }
}

@keyframes stars {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes astronaut {
  0%,
  100% {
    transform: translate(-50%, -50%) rotate(0deg) translateY(0);
  }
  25% {
    transform: translate(-50%, -50%) rotate(5deg) translateY(-10px);
  }
  50% {
    transform: translate(-50%, -50%) rotate(0deg) translateY(0);
  }
  75% {
    transform: translate(-50%, -50%) rotate(-5deg) translateY(10px);
  }
}

@keyframes arm-left {
  0%,
  100% {
    transform: rotate(15deg);
  }
  50% {
    transform: rotate(25deg);
  }
}

@keyframes arm-right {
  0%,
  100% {
    transform: rotate(-15deg);
  }
  50% {
    transform: rotate(-25deg);
  }
}
