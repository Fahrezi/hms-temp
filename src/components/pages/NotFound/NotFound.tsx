import type React from 'react';
import { Link } from 'react-router-dom';

import styles from './NotFound.module.scss';

const TITLE_DIGIT_CLASS = 'not_found__title-digit';

const NotFound: React.FC = () => {
  return (
    <div className={styles.not_found}>
      <div className={styles.not_found__container}>
        <h1 className={styles.not_found__title}>
          <span className={styles[TITLE_DIGIT_CLASS]}>4</span>
          <span className={styles[TITLE_DIGIT_CLASS]}>0</span>
          <span className={styles[TITLE_DIGIT_CLASS]}>4</span>
        </h1>

        <div className={styles.not_found__astronaut}>
          <div className={styles['not_found__astronaut-body']}></div>
          <div className={styles['not_found__astronaut-head']}></div>
          <div className={styles['not_found__astronaut-arm-left']}></div>
          <div className={styles['not_found__astronaut-arm-right']}></div>
          <div className={styles['not_found__astronaut-leg-left']}></div>
          <div className={styles['not_found__astronaut-leg-right']}></div>
        </div>

        <div className={styles.not_found__stars}></div>

        <div className={styles.not_found__message}>
          <h2 className={styles['not_found__message-title']}>Page Not Found</h2>
          <p className={styles['not_found__message-text']}>
            The page you are looking for might have been removed, had its name changed, or is temporarily unavailable.
          </p>
          <Link to='/' className={styles['not_found__message-button']}>
            Return to Dashobard
          </Link>
        </div>
      </div>
    </div>
  );
};

export default NotFound;
