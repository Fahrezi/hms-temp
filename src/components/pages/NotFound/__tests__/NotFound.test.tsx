import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import NotFound from '@/components/pages/NotFound';

describe('NotFound Component', () => {
  it('renders the 404 page correctly', () => {
    render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    expect(screen.getByText('Page Not Found')).toBeInTheDocument();
    expect(
      screen.getByText(
        'The page you are looking for might have been removed, had its name changed, or is temporarily unavailable.',
      ),
    ).toBeInTheDocument();
    expect(screen.getByText('Return to Dashobard')).toBeInTheDocument();
  });

  it('renders the 404 digits correctly', () => {
    render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    const digits = screen.getAllByText(/[404]/);
    expect(digits).toHaveLength(3);
  });

  it('renders the snapshot NotFound component', () => {
    const { container } = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });
});
