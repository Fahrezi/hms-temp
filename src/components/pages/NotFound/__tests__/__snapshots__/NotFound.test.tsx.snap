// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound Component > renders the snapshot NotFound component 1`] = `
<div>
  <div
    class="_not_found_b19e84"
  >
    <div
      class="_not_found__container_b19e84"
    >
      <h1
        class="_not_found__title_b19e84"
      >
        <span
          class="_not_found__title-digit_b19e84"
        >
          4
        </span>
        <span
          class="_not_found__title-digit_b19e84"
        >
          0
        </span>
        <span
          class="_not_found__title-digit_b19e84"
        >
          4
        </span>
      </h1>
      <div
        class="_not_found__astronaut_b19e84"
      >
        <div
          class="_not_found__astronaut-body_b19e84"
        />
        <div
          class="_not_found__astronaut-head_b19e84"
        />
        <div
          class="_not_found__astronaut-arm-left_b19e84"
        />
        <div
          class="_not_found__astronaut-arm-right_b19e84"
        />
        <div
          class="_not_found__astronaut-leg-left_b19e84"
        />
        <div
          class="_not_found__astronaut-leg-right_b19e84"
        />
      </div>
      <div
        class="_not_found__stars_b19e84"
      />
      <div
        class="_not_found__message_b19e84"
      >
        <h2
          class="_not_found__message-title_b19e84"
        >
          Page Not Found
        </h2>
        <p
          class="_not_found__message-text_b19e84"
        >
          The page you are looking for might have been removed, had its name changed, or is temporarily unavailable.
        </p>
        <a
          class="_not_found__message-button_b19e84"
          data-discover="true"
          href="/"
        >
          Return to Dashobard
        </a>
      </div>
    </div>
  </div>
</div>
`;
